src/main/java/org/jenkinsci/plugins/workflow/flow/StashManager.java

```diff
@@ -165,6 +167,35 @@ public static void copyAll(@Nonnull Run<?,?> from, @Nonnull Run<?,?> to) throws
         FileUtils.copyDirectory(fromStorage, storage(to));
     }
 
+    /**
+     * Get the filenames included in a particular stash.
+     * @param build The build to look in.
+     * @param stashName The stash name to look up.
+     * @return A possibly empty list of filenames included in the given stash name in the given build. Unix separators will be used.
+     * @throws IOException If unable to interact with the stash files.
+     */
+    @Nonnull
+    public static List<String> stashFileList(@Nonnull Run<?,?> build, @Nonnull String stashName) throws IOException {
+        List<String> contents = new ArrayList<>();
+        File[] kids = storage(build).listFiles();
+        if (kids != null) {
+            for (File kid : kids) {
+                String n = kid.getName();
+                if (n.endsWith(SUFFIX) && stashName.equals(n.substring(0, n.length() - SUFFIX.length()))) {
```

Comment:
Would be easier to look up the expected child name directly: `storage(build, stashName)`